calico/etcddriver/driver.py

```diff
@@ -512,11 +532,18 @@ def _handle_next_watcher_event(self, resync_in_progress):
         :raises FelixWriteFailed:
         :raises ResyncRequested:
         """
+        # PLW: What happens to the watcher thread if stop() gets called? Or does
+        # it plough on happily until the process goes down?
         if self._watcher_queue is None:
             raise WatcherDied()
         while not self._stop_event.is_set():
             # To make sure we always make progress, only trigger a new resync
             # if we're not in the middle of one.
+            # PLW: This means that when we get a kick to do a resync, we stop
+            # watching for a bit, so could miss an update or two. I think we
+            # should try to close that window, even though it's quite small.
+            # PLW: We could drop a resync because we are already doing one.
+            # Should at least catch and log that case.
```

Comment:
Suppose we fail to update an ipset with a transient error; in that case we'd fix it up when we next go round again (unless the transient error takes Felix down). Hence it feels like it has value, though it's not enormous value.

The biggest reason for leaving it turned on is that it means we exercise the code; code that is in but never on just rots. But you could argue we should nuke the code completely and we'd gain something.

My instinct is to say that we have it, so let's see if it causes issues and / or helps out. If we see lots of problems with it, we should retire it, and if it saves us from some bugs we should make sure we keep it on.